src/lib.rs

```diff
@@ -1861,37 +1712,55 @@ mod tests {
         event: *mut ffi::QUIC_CONNECTION_EVENT,
     ) -> QUIC_STATUS {
         let connection = unsafe { &*(context as *const Connection) };
-        let event = unsafe { (event as *const ConnectionEvent).as_ref().unwrap() };
-        match event.event_type {
-            crate::CONNECTION_EVENT_CONNECTED => {
+        let ev_ref = unsafe { event.as_ref().unwrap() };
+        let event = ConnectionEvent::from(ev_ref);
+        match event {
```

Comment:
This effectively means Rust event handlers have to go through two switch statements now?

Reply:
Unfortunately, yes. I hope the optimizer can do the job to eliminate one.